web/html/src/components/dialog/ReactDangerDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import { AsyncButton, Button } from "../buttons";
+import { Dialog, DialogProps } from "./ReactDialog";
+
+type Props = DialogProps & {
+  /** Text of the submit button */
+  submitText: string;
+  /** Icon of the submit button */
+  submitIcon: string;
+  /** Submit button class name */
+  btnClass?: string;
+  /** Data related to the dialog. This is passed as parameter to 'onConfirm*' handlers */
+  item?: any;
+  /** Function called when the submit button is clicked */
+  onConfirm?: (...args: any[]) => any;
+  /** Function called asynchronously when the submit button is clicked */
+  onConfirmAsync?: (...args: any[]) => Promise<any>;
+};
+
+/**
+ * A pop-up dialog for dangerous actions confirmation.
+ * It contains a 'Cancel' button and a button with text from 'submitText'
+ * and icon from 'submitIcon'.
+ * Related data may be passed with the 'item' property.
+ * This 'item' will be passed to the 'onConfirm*' handlers.
+ */
+export function DangerDialog(props: Props) {
+  const { item, btnClass, submitText, submitIcon, onConfirm, onConfirmAsync, ...otherProps}  = props;
+
+  const buttonClass = btnClass || "btn-danger";
+  const buttons = (
+    <div>
+      {props.onConfirmAsync ? (
+        <AsyncButton
+          text={submitText}
+          title={submitText}
+          icon={submitIcon}
+          defaultType={buttonClass}
+          action={() => {
+            onConfirmAsync?.(true);
+            if (props.onClose) {
+              props.onClose();
+            }
+          }}
+        />
+      ) : null}
+      {props.onConfirm ? (
+        <Button
+          className={buttonClass}
+          text={submitText}
+          title={submitText}
+          icon={submitIcon}
+          handler={() => {
+            onConfirm?.(item);
+            if (props.onClose) {
+              props.onClose();
```

Comment:
Can similarly to the previous line be `props.onClose?.()`, but I don't have a strong preference.

Reply:
nice catch